File: fibonacci/rust/src/bin/code.rs
use std::error::Error;

use fibonacci::*;

fn main() -> Result<(), Box<dyn Error>> {
    let args: Vec<String> = std::env::args().skip(1).collect();

    // try different integer sizes (u32/u64/u128/usize) to see their impact on performance (u8/u16 are too small)
    let runs: u32 = args.first().ok_or("missing number argument")?.parse()?;

    // sum the fibonacci numbers
    // (fold can infer the type automatically, sum needs explicit type annotations)
    #[allow(clippy::unnecessary_fold)]
    let sum = (0..runs).map(fibonacci).fold(0, |acc, n| acc + n);

    println!("{}", sum);

    Ok(())
}

File: fibonacci/zig/code.zig
const std = @import("std");

const fibonacci = @import("./fibonacci.zig").fibonacci;

pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = arena.allocator();

    const args_cli = try std.process.argsAlloc(allocator);

    if (args_cli.len < 2) {
        const stderr = std.io.getStdErr().writer();
        try stderr.writeAll("Please provide a number as argument.\n");
        std.process.exit(1);
    }

    const args = args_cli[1..];

    // try different integer sizes (u32/u64/u128/usize) to see their impact on performance
    const runs = try std.fmt.parseInt(u32, args[0], 0);

    const fn_fib = fibonacci(@TypeOf(runs));

    var sum: @TypeOf(runs) = 0;

    for (0..@intCast(runs)) |i| {
        sum += fn_fib(@intCast(i));
    }

    const stdout = std.io.getStdOut().writer();
    try stdout.print("{d}\n", .{sum});
}
